diff --git a/native/h264bsd_conceal.c b/native/h264bsd_conceal.c
index 97e924e..181d342 100644
--- a/native/h264bsd_conceal.c
+++ b/native/h264bsd_conceal.c
@@ -280,7 +280,7 @@ u32 ConcealMb(mbStorage_t *pMb, image_t *currImage, u32 row, u32 col,
     i32 firstPhase[16];
     i32 *pTmp;
     /* neighbours above, below, left and right */
-    i32 a[4] = { 0,0,0,0 }, b[4], l[4] = { 0,0,0,0 }, r[4];
+    i32 a[4] = { 0,0,0,0 }, b[4] = { 0,0,0,0 }, l[4] = { 0,0,0,0 }, r[4] = { 0,0,0,0 };
     u32 A, B, L, R;
 #ifdef H264DEC_OMXDL
     u8 fillBuff[32*21 + 15 + 32];
diff --git a/native/h264bsd_deblocking.c b/native/h264bsd_deblocking.c
index ec2bb08..c10b91a 100644
--- a/native/h264bsd_deblocking.c
+++ b/native/h264bsd_deblocking.c
@@ -708,7 +708,7 @@ void FilterVerLumaEdge(
                     tmp++;
                 }
 
-                val = (((q0 - p0) << 2) + (p1 - q1) + 4) >> 3;
+                val = ((4 * (q0 - p0)) + (p1 - q1) + 4) >> 3;
                 delta = CLIP3(-tmp, tmp, val);
 
                 p0 = clp[p0 + delta];
@@ -821,7 +821,7 @@ void FilterHorLumaEdge(
                 tmp++;
             }
 
-            val = ((((q0 - p0) << 2) + (p1 - q1) + 4) >> 3);
+            val = (((4 * (q0 - p0)) + (p1 - q1) + 4) >> 3);
             delta = CLIP3(-tmp, tmp, val);
 
             p0 = clp[p0 + delta];
@@ -900,7 +900,7 @@ void FilterHorLuma(
                     tmp++;
                 }
 
-                val = ((((q0 - p0) << 2) + (p1 - q1) + 4) >> 3);
+                val = (((4 * (q0 - p0)) + (p1 - q1) + 4) >> 3);
                 delta = CLIP3(-tmp, tmp, val);
 
                 p0 = clp[p0 + delta];
@@ -992,7 +992,7 @@ void FilterVerChromaEdge(
         if (bS < 4)
         {
             tc = thresholds->tc0[bS-1] + 1;
-            delta = CLIP3(-tc, tc, ((((q0 - p0) << 2) +
+            delta = CLIP3(-tc, tc, (((4 * (q0 - p0)) +
                       (p1 - q1) + 4) >> 3));
             p0 = clp[p0 + delta];
             q0 = clp[q0 - delta];
@@ -1015,7 +1015,7 @@ void FilterVerChromaEdge(
         if (bS < 4)
         {
             tc = thresholds->tc0[bS-1] + 1;
-            delta = CLIP3(-tc, tc, ((((q0 - p0) << 2) +
+            delta = CLIP3(-tc, tc, (((4 * (q0 - p0)) +
                       (p1 - q1) + 4) >> 3));
             p0 = clp[p0 + delta];
             q0 = clp[q0 - delta];
@@ -1068,7 +1068,7 @@ void FilterHorChromaEdge(
              ((u32)ABS(p1-p0) < thresholds->beta)  &&
              ((u32)ABS(q1-q0) < thresholds->beta) )
         {
-            delta = CLIP3(-tc, tc, ((((q0 - p0) << 2) +
+            delta = CLIP3(-tc, tc, (((4 * (q0 - p0)) +
                       (p1 - q1) + 4) >> 3));
             p0 = clp[p0 + delta];
             q0 = clp[q0 - delta];
@@ -1118,7 +1118,7 @@ void FilterHorChroma(
                  ((u32)ABS(p1-p0) < thresholds->beta)  &&
                  ((u32)ABS(q1-q0) < thresholds->beta) )
             {
-                delta = CLIP3(-tc, tc, ((((q0 - p0) << 2) +
+                delta = CLIP3(-tc, tc, (((4 * (q0 - p0)) +
                           (p1 - q1) + 4) >> 3));
                 p0 = clp[p0 + delta];
                 q0 = clp[q0 - delta];
diff --git a/native/h264bsd_decoder.h b/native/h264bsd_decoder.h
index ce05398..77fbf84 100644
--- a/native/h264bsd_decoder.h
+++ b/native/h264bsd_decoder.h
@@ -64,7 +64,7 @@ u32 h264bsdInit(storage_t *pStorage, u32 noOutputReordering);
 u32 h264bsdDecode(storage_t *pStorage, u8 *byteStrm, u32 len, u8 **picture, u32 *width, u32 *height);
 void h264bsdShutdown(storage_t *pStorage);
 
-storage_t* h264bsdAlloc();
+storage_t* h264bsdAlloc(void);
 void h264bsdFree(storage_t *pStorage);
 
 #endif /* #ifdef H264SWDEC_DECODER_H */
diff --git a/native/h264bsd_decoder.c b/native/h264bsd_decoder.c
index 57dbdc0..e8d4d2b 100644
--- a/native/h264bsd_decoder.c
+++ b/native/h264bsd_decoder.c
@@ -93,8 +93,8 @@ u32 h264bsdInit(storage_t *pStorage, u32 noOutputReordering)
     /* allocate mbLayer to be next multiple of 64 to enable use of
      * specific NEON optimized "memset" for clearing the structure */
     size = (sizeof(macroblockLayer_t) + 63) & ~0x3F;
 
-    pStorage->mbLayer = (macroblockLayer_t*)malloc(size);
+    pStorage->mbLayer = (macroblockLayer_t*)H264BSD_MALLOC(size);
     if (!pStorage->mbLayer)
         return HANTRO_NOK;
 
@@ -498,7 +498,6 @@ u32 h264bsdDecodeInternal(storage_t *pStorage, u8 *byteStrm, u32 len,
 }
 
 u32 h264bsdDecode(storage_t *pStorage, u8 *byteStrm, u32 len, u8 **picture, u32 *width, u32 *height) {
-  u32 bytesRead = 0;
   u32 retCode = 3;
   u32 readBytes = 0;
 
@@ -596,7 +595,7 @@ void h264bsdShutdown(storage_t *pStorage)
 
-storage_t* h264bsdAlloc()
+storage_t* h264bsdAlloc(void)
 {
-    return (storage_t*)malloc(sizeof(storage_t));
+    return (storage_t*)H264BSD_MALLOC(sizeof(storage_t));
 }
 
 /*------------------------------------------------------------------------------
@@ -619,5 +618,5 @@ storage_t* h264bsdAlloc()
 
 void h264bsdFree(storage_t *pStorage)
 {
-    free(pStorage);
+    H264BSD_FREE(pStorage);
 }
diff --git a/native/h264bsd_pic_order_cnt.c b/native/h264bsd_pic_order_cnt.c
index 33d781d..65d5626 100644
--- a/native/h264bsd_pic_order_cnt.c
+++ b/native/h264bsd_pic_order_cnt.c
@@ -85,8 +85,8 @@ i32 h264bsdDecodePicOrderCnt(pocStorage_t *poc, seqParamSet_t *sps,
 
     u32 i;
     i32 picOrderCnt;
-    u32 frameNumOffset, absFrameNum, picOrderCntCycleCnt;
-    u32 frameNumInPicOrderCntCycle;
+    u32 frameNumOffset, absFrameNum, picOrderCntCycleCnt = 0;
+    u32 frameNumInPicOrderCntCycle = 0;
     i32 expectedDeltaPicOrderCntCycle;
     u32 containsMmco5;
 
diff --git a/native/h264bsd_reconstruct.c b/native/h264bsd_reconstruct.c
index 496a5df..af3cded 100644
--- a/native/h264bsd_reconstruct.c
+++ b/native/h264bsd_reconstruct.c
@@ -2257,7 +2257,6 @@ void h264bsdFillBlock(
 /* Variables */
 
     i32 xstop, ystop;
-    void (*fp)(u8*, u8*, i32, i32, i32);
     i32 left, x, right;
     i32 top, y, bottom;
 
@@ -2274,13 +2273,6 @@ void h264bsdFillBlock(
     xstop = x0 + (i32)blockWidth;
     ystop = y0 + (i32)blockHeight;
 
-    /* Choose correct function whether overfilling on left-edge or right-edge
-     * is needed or not */
-    if (x0 >= 0 && xstop <= (i32)width)
-        fp = FillRow1;
-    else
-        fp = h264bsdFillRow7;
-
     if (ystop < 0)
         y0 = -(i32)blockHeight;
 
diff --git a/native/h264bsd_util.h b/native/h264bsd_util.h
index a61095b..a9cfebf 100644
--- a/native/h264bsd_util.h
+++ b/native/h264bsd_util.h
@@ -36,6 +36,7 @@
 #include <stdint.h>
 #include "basetype.h"
 #include "h264bsd_cfg.h"
+#include "h264bsd_port.h"
 #include "h264bsd_stream.h"
 #include "h264bsd_image.h"
 
@@ -141,13 +142,13 @@
 /* macro to allocate memory */
 #define ALLOCATE(ptr, count, type) \
 { \
-    (ptr) = malloc((count) * sizeof(type)); \
+    (ptr) = H264BSD_MALLOC((count) * sizeof(type)); \
 }
 
 /* macro to free allocated memory */
 #define FREE(ptr) \
 { \
-    free((ptr)); (ptr) = NULL; \
+    H264BSD_FREE((ptr)); (ptr) = NULL; \
 }
 
 #define ALIGN(ptr, bytePos) \
